### ISSUE TITLE
MvcMailer FileLoadExcpetion in MVC4

### ISSUE THREAD

**Issue by ahordijk** [smsohan/MvcMailer#42]
When using MvcMailer in the Cloud and publishing the package the state of the publish stays in cycling.
The Solution is created with MVC4. After numerous attemps intellitrace noted that it could not find / load System.Web.Mvc version 3. 

After troubleshooting all included nuget packeges in the solution we commented all MvcMailer specific code (its base class and implementation). At this point the application can be hosted in the cloud with MVC4 but without the MvcMailer :-(

My question is do you reference explicitly to System.Web.Mvc version 3 in your code? If so can you change this so MvcMailer can be used with MVC4? 

Steps to reproduce:
1.  Create new MVC 4 beta project
2.  Add nuget package Mvc Mailer. Optional: update the default packages or add other packages.
3.  Project won’t run properly because the web.config in the Views folder has double config sections so comment them:

1.  Add class that implements te base class for the mvc mailer:
   public class TestMailer : MailerBase
   {
   public TestMailer(): base()
   {
             MasterName="_Layout";
      }
   }
2.  Publish to Azure with default settings. The state in Visual Studio 2010 will stay in Cycle. The intellitrace log file says: System.TypeLoadException. Could not load file or assembly MVC version 3.

**Comment by smsohan** [comment]
I need to check this out. Is this a problem that you see on the cloud only?

---

Sent from my iPhone
S M Sohan

On 2012-03-15, at 4:38 AM, ahordijk
reply@reply.github.com
wrote:

**Comment by ahordijk** [comment]
Yes local on my developing pc this problem does not occur. Hope you can solve this One. 
This is because MVC3 is also installed on my development pc.

----- Reply message -----
Van: "S" reply@reply.github.com
Aan: "ahordijk" arjan.hordijk@gmail.com
Onderwerp: [MvcMailer] MvcMailer FileLoadExcpetion in MVC4 (#42)
Datum: do, mrt. 15, 2012 14:51

I need to check this out. Is this a problem that you see on the cloud only?

---

Sent from my iPhone
S M Sohan

On 2012-03-15, at 4:38 AM, ahordijk
reply@reply.github.com
wrote:

> When using MvcMailer in the Cloud and publishing the package the state of the publish stays in cycling.
> The Solution is created with MVC4. After numerous attemps intellitrace noted that it could not find / load System.Web.Mvc version 3.
> 
> After troubleshooting all included nuget packeges in the solution we commented all MvcMailer specific code (its base class and implementation). At this point the application can be hosted in the cloud with MVC4 but without the MvcMailer :-(
> 
> My question is do you reference explicitly to System.Web.Mvc version 3 in your code? If so can you change this so MvcMailer can be used with MVC4?
> 
> Steps to reproduce:
> 1.    Create new MVC 4 beta project
> 2.    Add nuget package Mvc Mailer. Optional: update the default packages or add other packages.
> 3.    Project won’t run properly because the web.config in the Views folder has double config sections so comment them:
> 
> 
> 
> 
> 1.    Add class that implements te base class for the mvc mailer:
>    public class TestMailer : MailerBase
>    {
>    public TestMailer(): base()
>    {
>             MasterName="_Layout";
>      }
>    }
> 2.    Publish to Azure with default settings. The state in Visual Studio 2010 will stay in Cycle. The intellitrace log file says: System.TypeLoadException. Could not load file or assembly MVC version 3.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/smsohan/MvcMailer/issues/42

---

**Comment by LockTar** [comment]
Is this bug fixed or is there an estimated time that it is fixed?
Thank you.

**Comment by LockTar** [comment]
@smsohan Is there some more information about this?

**Comment by LockTar** [comment]
Any update available?

**Comment by ahordijk** [comment]
@smsohan Can you give me any feedback if you are going to fix this one? 
Or a roadmap / planning for when you will support mvc4?

**Comment by JoshReedSchramm** [comment]
Ran into this today - sent pull request -> https://github.com/smsohan/MvcMailer/pull/54 to address.

**Comment by smsohan** [comment]
Thanks for this. I haven't been able to spend much time on MvcMailer off
late due to my lack of touch with .Net project for the last couple of
years. But I will try this out over the weekend and if everything looks

## cool, I will push an update.

Sohan
http://smsohan.com
skype:smsohan | gtalk:sohan39 | cell: 403-714-2673

**Comment by JoshReedSchramm** [comment]
Cool. I also noticed the web.config in Views is writing wrong, going to look at that next. I think there's another issue. 

I didn't generate a new nuget package in source control, i did locally to test but I didn't want to impose a new version or anything so you may need to do that.

**Comment by mikecousins** [comment]
I'm currently getting hosed by this. I checked the pull request and it looks pretty solid to me.

**Comment by justinsoliz** [comment]
I agree any chance we can get this pull request accepted?
